transport/server.cc

```
@@ -1694,21 +1751,34 @@ cql_server::connection::make_schema_change_event(const event::schema_change& eve
 void cql_server::connection::write_response(foreign_ptr<std::unique_ptr<cql_server::response>>&& response, service_permit permit, cql_compression compression)
 {
     _ready_to_respond = _ready_to_respond.then([this, compression, response = std::move(response), permit = std::move(permit)] () mutable {
-        auto message = response->make_message(_version, compression);
-        message.on_delete([response = std::move(response)] { });
-        return _write_buf.write(std::move(message)).then([this] {
+        utils::result_with_eptr<scattered_message<char>> message = response->make_message(_version, compression);
+        if (!message) [[unlikely]] {
+            if (auto* exp = try_catch<exceptions::cassandra_exception>(message.error())) {
+                clogger.error("{}: write_response: failed to make message, code {}, message [{}]", _client_state.get_remote_address(), exp->code(), exp->what());
+            } else if (auto* exp = try_catch<std::exception>(message.error())) {
+                clogger.error("{}: write_response: failed to make message, message [{}]", _client_state.get_remote_address(), exp->what());
+            } else {
+                clogger.error("{}: write_response: failed to make message", _client_state.get_remote_address());
+            }
+            return make_ready_future<>();
+        }
+        message.value().on_delete([response = std::move(response)] { });
+        return _write_buf.write(std::move(message).value()).then([this] {
```

You have snuck in a logic change into a refactoring commit. Please move it out to a separate commit.

By the way, why the change? Why do we swallow the exception now rather than propagating it outwards?

---

I'll extract `make_message` and `make_frame` changes into a separate commit. As for the logic change, I see, I should keep it the same. There seem to be 13 places in `transport/event_notifier.cc` that call `write_response`. I'll propagate the error further up without rethrowing for both transport server and transport event notifier.